### PR TITLE
[YUNIKORN-1701] Make simple preemptor cluster agnostic

### PR DESCRIPTION
**Pull request by manirajv06**
### What is this PR for?
Ensure simple preemptor runs seamlessly in all cluster environments.

### What type of PR is it?
* [ ] - Bug Fix

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1701

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: test/e2e/simple_preemptor/simple_preemptor_test.go [hunk 1]
```diff
@@ -79,13 +81,29 @@ var _ = ginkgo.BeforeSuite(func() {
 
 	// Extract node allocatable resources
 	for _, node := range nodes.Items {
-		if node.Name == Worker1 {
+		// skip master if it's marked as such
+		node := node
+		if k8s.IsMasterNode(&node) || !k8s.IsComputeNode(&node) {
```
Comment (craigcondit) [inline test/e2e/simple_preemptor/simple_preemptor_test.go:86]: Isn't any node which is not a master node by definition a compute node? Why the additional check?

File: test/e2e/simple_preemptor/simple_preemptor_test.go [hunk 2]
```diff
@@ -79,13 +81,29 @@ var _ = ginkgo.BeforeSuite(func() {
...
+			continue
+		}
+		if Worker1 == "" {
```
Comment (craigcondit) [inline test/e2e/simple_preemptor/simple_preemptor_test.go:89]: This code seems brittle. Can we just iterate the nodes and take the first two matching candidates without all the special-casing?

**Review by craigcondit — CHANGES_REQUESTED** [review]
See comments.

File: test/e2e/simple_preemptor/simple_preemptor_test.go (hunk 1 above)
Comment (manirajv06) [reply test/e2e/simple_preemptor/simple_preemptor_test.go:86]: I have seen 3 types - 1. Master 2. Workers 2a). Infra 2b). Compute. We have to consider only compute nodes, not even 2a for running pods. Once Compute nodes has been chosen, we have to limit the no. of compute nodes (mostly two) for running pods and taint other compute nodes as un schedulable based on the use cases.

File: test/e2e/framework/helpers/k8s/k8s_utils.go [hunk 3]
```diff
@@ -1292,10 +1292,22 @@ func IsMasterNode(node *v1.Node) bool {
...
+	roleNodeLabelExists := false
+	for labelKey, labelValue := range node.Labels {
+		if labelKey == common.RoleNodeLabel {
+			roleNodeLabelExists = true
+			if _, ok := common.ComputeNodeLabels[labelValue]; ok {
+				return true
+			}
+		}
+	}
+	return !roleNodeLabelExists
```
[the remark this answers is not in the document]
Comment (manirajv06) [reply test/e2e/framework/helpers/k8s/k8s_utils.go:1308]: There is no "role" node label in kind clusters setup. "role" node label is available in aks/eks clsusters. IsMasterNode method  is good enough for Kind clusters to separate out the master from worker nodes. IsComputeNode is mainly for separate out the compute worker nodes from infra worker nodes in aks/eks clusters. Just to satisfy kind clusters or any other cluster setup (where there is no "role" node label), we need to return true here (!roleNodeLabelExists).

File: test/e2e/simple_preemptor/simple_preemptor_test.go (hunk 2 above)
Comment (manirajv06) [reply test/e2e/simple_preemptor/simple_preemptor_test.go:89]: We need to exclude 1. Master 2. Infra worker nodes and choose only "compute" worker nodes. Even no. of "compute" worker nodes (mostly two) would be chosen based on use cases and other remaining "compute" worker nodes would be tainted as "unschedulable" till we complete the test execution.

File: test/e2e/simple_preemptor/simple_preemptor_suite_test.go [hunk 4]
```diff
@@ -36,17 +35,16 @@ func init() {
 
 func TestSimplePreemptor(t *testing.T) {
 	ginkgo.ReportAfterSuite("TestSimplePreemptor", func(report ginkgo.Report) {
-		err := common.CreateJUnitReportDir()
-		Ω(err).NotTo(gomega.HaveOccurred())
-		err = reporters.GenerateJUnitReportWithConfig(
+		err := reporters.GenerateJUnitReportWithConfig(
 			report,
-			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-simple_preemptor_junit.xml"),
+			filepath.Join(configmanager.YuniKornTestConfig.LogDir, "TEST-simple_pre_emptor_junit.xml"),
```
Comment (craigcondit) [inline test/e2e/simple_preemptor/simple_preemptor_suite_test.go:40]: Minor nit: Preemptor is one word. WIll fix this on commit.

**Review by craigcondit — APPROVED** [review]
+1.

**Pull request closed without merging**